sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -734,9 +734,13 @@ impl SocRegistersImpl {
 
         if self.timer.fired(&mut self.op_fw_read_complete_action) {
             // Receiver sets status as CMD_COMPLETE after reading the mailbox data.
-            if !self.mailbox.is_status_cmd_busy() {
+            if !self.mailbox.regs().status().read().status().cmd_busy() {
```

Comment:
Polling for busy is part of the sender protocol. We should change the comment and use soc_mailbox here.

Reply:
Added a followup commit to change this (I don't like to mix refactoring and functional changes in the same PR).